test/swarm.js

```diff
@@ -174,36 +179,37 @@ test('one server, one client - banned peer does not reconnect', async (t) => {
 })
 
 test('two servers, two clients - simple deduplication', async (t) => {
+  const connection1Test = t.test('connection1')
+  const connection2Test = t.test('connection2')
+
+  connection1Test.plan(1)
+  connection2Test.plan(1)
+
   const { bootstrap } = await createTestnet(3, t.teardown)
 
   const swarm1 = new Hyperswarm({ bootstrap, backoffs: BACKOFFS, jitter: 0 })
   const swarm2 = new Hyperswarm({ bootstrap, backoffs: BACKOFFS, jitter: 0 })
 
-  let s1Connections = 0
-  let s2Connections = 0
+  t.teardown(async () => {
+    await swarm1.destroy()
+    await swarm2.destroy()
+  })
 
   swarm1.on('connection', (conn) => {
-    s1Connections++
+    connection1Test.pass('Swarm 1 connection')
     conn.on('error', noop)
   })
   swarm2.on('connection', (conn) => {
-    s2Connections++
+    connection2Test.pass('Swarm 2 connection')
     conn.on('error', noop)
   })
 
   const topic = Buffer.alloc(32).fill('hello world')
   await swarm1.join(topic).flushed()
   await swarm2.join(topic).flushed()
 
-  await swarm2.flush()
-  await swarm1.flush()
-  await timeout(250)
-
-  t.is(s1Connections, 1)
-  t.is(s2Connections, 1)
-
-  await swarm1.destroy()
-  await swarm2.destroy()
+  await flushConnections(swarm2)
```

Comment:
Same as above: pretty sure these 2 flushes are superfluous because of how you modified the test, having put the assertions inside the on-connection handlers

Using the same reasoning: you can also replace the previous line (`await swarm2.join(topic).flushed()`) by `await swarm2.join(topic)`

Reply:
fixed